### DISCUSSION TITLE
 Can I clone snapshot data to my bucket through S3

### DISCUSSION THREAD

**Discussion by J1a-wei** [filecoin-project/lotus#8824] [General Q&A] [answered]
Hi, I has deployed lotus in my eks.  To reduce costs, I want to copy snapshot into my s3 . 
<img width="1873" alt="image" src="https://user-images.githubusercontent.com/49090950/172573665-a6493709-abf9-4c8f-8614-70882775ee22.png">
I'm not sure this scenario is feasible.It looks like the URL is the redirection to the latest Snapshot

```
$ curl -i https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car
HTTP/1.1 200 OK
x-amz-id-2: S3Fr+I0GFmY1X3JgEn+kshev1h6XmlqLqaBq5JEvvwFqrDh9wbBlP82kdIxgdAMjbyOVm4L55ck=
x-amz-request-id: XJJF8KK505CRJ49C
Date: Wed, 08 Jun 2022 08:51:03 GMT
Last-Modified: Wed, 08 Jun 2022 08:04:28 GMT
ETag: "fab82f5ac6df8f774494e9bee582ee8f-9177"
x-amz-website-redirect-location: https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_1878720_2022-06-08_06-00-00.car
Accept-Ranges: bytes
Content-Type: binary/octet-stream
Server: AmazonS3
Content-Length: 76981008243

```

It is impossible to judge where the Filecoin Snapshot S3 region is through the URL. I am not sure if I will encounter problems such as permissions. Can you give me some tips?

**Comment by LexLuthr** [comment, marked as answer]
@skrbug AWS does not allow hardware to connect across zones but that is the URL to an object. It doesn't require permission. 
You can simply use the documented steps to download the snapshot.

**Reply by J1a-wei** [reply]
Yes I Can download and then upload my s3. Can I let AWS clone object which lotus provider into my s3?

**Reply by LexLuthr** [reply]
Why do you want to upload to your own S3? Multiple snapshots are generated daily. If you clone all these objects you will end up needing 10s of TBs of space in a matter of days.

**Reply by J1a-wei** [reply]
To reduce data transfer costs. lotus download these file from same region which my private s3. We have a lots of  pods which run lotus. So It's just my thought. If I can cache snapshot data in my private s3 and edit lotus env (snapshot url). Maybe it will reduce costs.
But I can't find the snapshot s3 uri.  
aws s3 cp ref
https://docs.aws.amazon.com/cli/latest/userguide/cli-services-s3-commands.html#using-s3-commands-managing-objects-copy

thanks in advance !

**Reply by LexLuthr** [reply]
I cannot comment on the AWS part. You will need to check their docs for this. 
Why do you want to run lots of lotus pod? What is your purpose behind running lotus? Maybe, I can help clarify things.

**Reply by J1a-wei** [reply]
By the way. I don't clone all the data, I only clone the latest data of the day, and clean the old data regularly

**Reply by J1a-wei** [reply]
We are making a product for developers.   The point is that I can't find where the bucket which provide snapshot content

**Reply by LexLuthr** [reply]
@travisperson Can you help with this ask? They want to copy out snapshot from bucket as soon as it is available.

**Reply by travisperson** [reply]
Not with the `fil-chain-snapshots-fallback` bucket. There is no support for that bucket. We are working on a new service to provide snapshots through s3. Once that is launched this issue can be reopened and I can possibly assists.

**Reply by J1a-wei** [reply]
Thank you for your answers @travisperson @LexLuthr 

We are looking forward to this feature!

**Comment by LexLuthr** [comment]
@skrbug I am archiving this discussion for now. Once we have new service in place, we can open a new discussion for technicals to copy the objects from that bucket.
